The expression was reversed. Without this it compiles but
the compat module will fail to load since a symbol
dma_common_get_sgtable is already owned by the kernel.

This should be applied to the linux-3.11.y branch.

Signed-off-by: Luis R. Rodriguez <mcgrof@do-not-panic.com>

Further fixups to work on 3.5 kernels.

Signed-off-by: Robert Hooker <sarvatt@ubuntu.com>
---
 backports/compat/Kconfig |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/backports/compat/Kconfig b/backports/compat/Kconfig
index 8377013..8d51b50 100644
--- a/backports/compat/Kconfig
+++ b/backports/compat/Kconfig
@@ -170,7 +170,7 @@ config BACKPORT_BUILD_DMA_SHARED_HELPERS
 	depends on HAS_DMA
 	# Build on other kernels kernels < 3.9 if HAVE_GENERIC_DMA_COHERENT is
 	# not set. Kernels >= 3.8 have this if HAS_DMA is set.
-	depends on (!HAVE_GENERIC_DMA_COHERENT || BACKPORT_KERNEL_3_9)
+	depends on (!HAVE_GENERIC_DMA_COHERENT && BACKPORT_KERNEL_3_9)
 	# Always build if on 3.3 - 3.5
 	default y if (BACKPORT_KERNEL_3_4 || BACKPORT_KERNEL_3_5 || BACKPORT_KERNEL_3_6)
 	# build for testing
-- 
1.7.10.4
